### DISCUSSION TITLE
Add a separate flatpak for avx2 version

### DISCUSSION THREAD

**Discussion by code-IM-perfect** [zen-browser/desktop#2089] [Ideas]
Currently the only flatpak available is the generic version, however avx2 is supported on a vast majority of modern processors, it should not be very hard to implement as there is already a maintained AppImage and Portable Version. I wonder why its not already implemented

**Comment by nocanoa** [comment]
Mainly because flathub doesnt alow 2 versions, but we already chating about it on discord

**Reply by code-IM-perfect** [reply]
what are the possible solutions to this? what exactly is being discussed?

**Reply by nocanoa** [reply]
there is no current gh action for opt flatpak, (dont use linux/flatpak so my knoledg is limited), not 100% sure how flatpack can have auto updates if it fetches it from a storage server (witch I have) if anyone wants to make a gh action that builds it, and upload it to the server to try if it works letmeknow 😉

**Reply by latenightdef** [reply]
I've seen an app with 2 versions both on Flathub that is Qalculate!. They have a [GTK version](https://github.com/flathub/io.github.Qalculate/blob/master/io.github.Qalculate.yaml) and a [Qt version](https://github.com/flathub/io.github.Qalculate.qalculate-qt/blob/master/io.github.Qalculate.qalculate-qt.yaml) that are separated into 2 Flatpak manifests and that means 2 different packages from the same developer on Flathub.

We can probably achieve this in a similar way by creating one for the generic build and another one for optimized build. (with different source archive URLs)

**Comment by RealDishwash** [comment]
Most of the people using flatpak are going to be people using modern processors so i think it makes sense, the people who have older processors are more likely to use appimage/source.

**Comment by Azarilh** [comment]
Exactly. Most people would have no problems using avx2 version. Could maybe implement a quick check upon first startup of the app, that checks whether the PC supports avx2 or not, and if it doesn't, tell the user with a popup and tell them to install the AppImage.

**Reply by Sinaaaa** [reply]
> Exactly. Most people would have no problems using avx2 version. Could maybe implement a quick check upon first startup of the app, that checks whether the PC supports avx2 or not, and if it doesn't, tell the user with a popup and tell them to install the AppImage.

Just to represent the opposing view. On my beloved thinkpad I would just revert back to Floorp if you all made the flatpak avx2 only. The appimage is not an alternative everyone is willing to accept, I'm certainly not. 

>Most of the people using flatpak are going to be people using modern processors so i think it makes sense

I don't see how flatpak usage would mean more modern systems on average than otherwise.

**Reply by RealDishwash** [reply]
> > Exactly. Most people would have no problems using avx2 version. Could maybe implement a quick check upon first startup of the app, that checks whether the PC supports avx2 or not, and if it doesn't, tell the user with a popup and tell them to install the AppImage.
> 
> Just to represent the opposing view. On my beloved thinkpad I would just revert back to Floorp if you all made the flatpak avx2 only. The appimage is not an alternative everyone is willing to accept, I'm certainly not.
> 

It's been supported on intel CPU's since 2013 and AMD CPU's since 2015 so anything from the last 10 years should support it. If you look at the steam hardware survey, 94.44% of tested CPU's support it.

[AVX2 (94.44%)](https://store.steampowered.com/hwsurvey)
> > Most of the people using flatpak are going to be people using modern processors so i think it makes sense
> 
> I don't see how flatpak usage would mean more modern systems on average than otherwise.

On average, the users of older hardware tend to stick to distro packaging, appimages, the aur and tarballs than using flatpak because of "bloat".

 I understand that there's probably still some users such as yourself that still use it though, perhaps we could default the flathub build to avx2 and then have a separate flatpak remote for generic?

**Comment by Sinaaaa** [comment]
>[AVX2 (94.44%)](https://store.steampowered.com/hwsurvey)

That's a mostly a Windows Computer related survey, people tend to throw away perfectly viable hardware a lot less frequently in the Linux space. Some of the most beloved & most robustly built Thinkpads like the T420-520 don't support AVX2 & they are still popular among Linux users, especially outside of the States.

>On average, the users of older hardware tend to stick to distro packaging, appimages, the aur and tarballs than using flatpak because of "bloat".

Maybe this has a statistical significance due to sentiment, but on hardware that is strong enough to browse the modern internet -core 2 duo and up- the so called bloat will not mean much, or at least not anymore than on a modern system, which would mean 1-2 extra seconds of load time from an ssd.

**Reply by RealDishwash** [reply]
What do you think of having a different flatpak remote for the generic package

**Reply by Sinaaaa** [reply]
> What do you think of having a different flatpak remote for the generic package

It wouldn't be great, but maybe I wouldn't stop using Zen on this laptop in that eventuality. 

How would you feel if the AVX3 package were the go to the external flatpak remote instead? I imagine you wouldn't be very happy either ^
I imagine it's kind of a 5% of users vs. 5% performance situation. It would be nice if somehow this could work out within Flathub.

**Reply by code-IM-perfect** [reply]
I mean if there was an AVX2 flatpak on a different remote, I would not be complaining
